implementation/src/main/java/io/smallrye/health/checks/InetAddressHealthCheck.java

```diff
@@ -0,0 +1,69 @@
+package io.smallrye.health.checks;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+
+/**
+ * Health check implementation to check if host is reachable using {@code java.net.InetAddress.isReachable} method.
+ *
+ * <pre>
+ * {@code
+ * @Produces
+ * @ApplicationScoped
+ * @Liveness
+ * HealthCheck check1() {
+ *   return new InetAddressHealthCheck("service.com");
+ * }
+ * }
+ * </pre>
+ *
+ * @see InetAddress
+ */
+public class InetAddressHealthCheck implements HealthCheck {
+
+    private static final String DEFAULT_NAME = "Internet Address Check";
+    private static final int DEFAULT_TIMEOUT = 2000;
+
+    private String host;
+    private String name;
+    private int timeout;
+
+    public InetAddressHealthCheck(String host) {
+        this.host = host;
+        this.name = DEFAULT_NAME;
```

Comment:
Is there any difference between setting the field value in a constructor vs setting the field directly? Mostly curious if there is one

Reply:
I think there is no difference, apart from when they are initialized. Said that I usually prefer this approach because you as a developer usually inspect the constructor not directly the fields, so following the constructor (or debugging it) you see all the time what are the default values.